docs/rfcs/1013-parsec-auth-scheme.md

```diff
@@ -0,0 +1,267 @@
+      "encrypted_priv_key_key_algorithm": "KeyAlgorithm"
+    }
+  }
+}
+```
+
+`EncryptedAccountManifest`, `EncryptedSymKey`, `HMACKey`, `EncryptedPrivKey` are specialized `Bytes` types.
+
+`EncryptedAccountManifest` is generated by encrypting the following rust structure:
+
+```rust
+struct AccountManifest {
+  /// List of registered auth medium for the account.
+  auth_mediums: HashMap<AuthMediumID, PublicKey>,
+  /// List of devices registered for the account.
+  device_keys: HashMap<(OrganizationID, DeviceID), SymmetricKey>,
+}
+```
+
+### Obtain the account manifest
+
+Before obtaining the encrypted account manifest, the user retrieves the information related to its authentication medium:
+
+```yml
+{
+  "cmd": "auth_medium_get",
+  "req": {
+    "id": "AuthMediumID"
+  },
+  "reps": [
+    {
+      "status": "ok",
+      "account_manifest": "EncryptedAccountManifest",
+      "encrypted_priv_key_key_algorithm": "KeyAlgorithm",
+      "encrypted_priv_key": "EncryptedPrivKey"
+    }
+  ]
+}
+```
+
+And now request the account manifest:
+
+```yml
+{
+  "cmd": "account_manifest_get",
+  "req": {},
+  "status": [
+    {
+      "status": "ok",
+      "account_manifest": "EncryptedAccountManifest"
+    }
+  ]
+}
+```
+
+### Uploading a new device
+
+Now that the client can encrypt the device with the symmetric key, it can upload it:
+
+```yml
+{
+  "cmd": "device_upload",
+  "req": {
+    "account_manifest": "EncryptedAccountManifest",
+    "organization_id": "OrganizationID",
+    "device_ownership_proof": "DeviceOwnershipProof",
+    "encrypted_device": "EncryptedDevice"
+  },
+  "reps": [
+    {
+      "status": "ok"
+    },
+    {
+      "status": "auth_medium_mismatch"
+    },
+    {
+      "status": "invalid_ownership_proof"
+    }
+  ]
+}
+```
+
+> [!NOTE]
+> The user can upload multiple devices for the same organization.
+
+### List available devices
+
+To list the devices registered in the service, the client only needs to be authenticated:
+
+```yml
+{
+  "cmd": "device_list",
+  "req": {},
+  "reps": [
+    {
+      "status": "ok",
+      "devices": "Vec<DeviceInfo>"
+    }
+  ],
+  "custom_types": {
+    "DeviceInfo": {
+      "organization_id": "OrganizationID",
+      "device_id": "DeviceID",
+      "user_id": "UserID",
+      "human_handle": "HumanHandle",
+      "created_on": "DateTime",
+      "revoked_on": "Option<DateTime>"
+    }
+  }
+}
+```
+
+### Retrieve device
+
+To retrieve the encrypted device, the client simply has to provide the organization ID:
+
+```yml
+{
+  "cmd": "device_get",
+  "req": {
+    "organization_id": "OrganizationId",
+    "device_id": "DeviceID"
+  },
+  "reps": [
+    {
+      "status": "ok",
+      "encrypted_device": "EncryptedDevice"
+    },
+    {
+      "status": "device_not_found"
+    }
+  ]
+}
+```
+
+### Adding a new authentication method
+
+To add a new authentication method, the client just needs to provide the required information:
+
+```yml
+{
+  "cmd": "auth_medium_new",
```

Comment:
```suggestion
  "cmd": "auth_method_add",
```
or to be consistent with existing commands (`realm_create`, `user_create`, etc.)
```suggestion
  "cmd": "auth_method_create",
```